jupyterhub/templates/NOTES.txt

```diff
@@ -48,4 +48,16 @@ resources.
 DEPRECATION: hub.uid is deprecated in jupyterhub chart 0.9. Set the hub.containerSecurityContext.runAsUser value
 directly instead.
 
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{- if hasKey .Values.hub "imagePullSecret" }}
+{{- if .Values.hub.imagePullSecret.enabled }}
+{{ "HARD DEPRECATION: hub.imagePullSecret has renamed to global.imagePullSecret" | fail }}
+{{- end }}
+{{- end }}
+
+{{- if hasKey .Values.singleuser "imagePullSecret" }}
+{{- if .Values.singleuser.imagePullSecret.enabled }}
+{{ "HARD DEPRECATION: singleuser.imagePullSecret has renamed to global.imagePullSecret" | fail }}
+{{- end }}
+{{- end }}
```

Comment:
This is the breaking changes that the user will be aware of if needed during Helm template rendering. This is because when this template renders, the `| fail` part will emit the message piped to it.

I've considered a sot deprecation, but the Helm templates logic to allow for that is messy, and we don't want to silently let users get stuck failing to pull images suddenly because their configuration isn't working any more, then its better to fail hard before any changes are made to k8s.